src/java.base/share/classes/java/lang/reflect/code/op/CoreOps.java

```diff
@@ -2544,7 +2744,12 @@ public NegOp transform(CopyContext cc, OpTransformer ot) {
         }
 
         NegOp(TypeDesc resultType, Value v) {
-            super(NAME, resultType, v);
+            super(NAME, v);
+        }
+
+        @Override
+        public TypeDesc resultType() {
+            return operands().get(0).type();
         }
```

Comment:
Move to `UnaryOp`?

Reply:
It can be moved, I just wasn't sure if the pattern will hold for every UnaryOp.